TLM/TLM/Manager/Impl/TrafficPriorityManager.cs

```diff
@@ -431,7 +440,8 @@ public class TrafficPriorityManager
                 return true;
             }
 
-            if ((vehicle.m_flags & Vehicle.Flags.Emergency2) != 0) {
+            if ((vehicle.m_flags & Vehicle.Flags.Emergency2) != 0 &&
+                vehicle.Info.m_vehicleType.IsFlagSet(VehicleInfo.VehicleType.Car)) {
```

Comment:
I would not assume that, sadly. I did a lot of tests in-game and if you don't explicitly tell to the runtime that method _can be_ inlined, it won't even try that on Windows (I've tested super short methods).

Reply:
You can suggest inline with method attribute: `[MethodImpl(1 << 8)]` (undocumented in `mscorlib.dll`, but mono runtime compiler knows what to do with it)

_Once added it will become impossible to patch with harmony if Jit decide to inline it (plus may crash debugger when you try to step into method with that attribute)._

I also added a task to my todo list to check methods flow and see where we could refactor things to pass structs by ref instead of spamming with `ToXyz()` extensions.

![image](https://user-images.githubusercontent.com/19638970/162038094-6341dbf3-cc2f-445b-98a4-fd1b9ea2d293.png)

![image](https://user-images.githubusercontent.com/19638970/162038128-686fa90b-436c-480f-a73c-56bf2dd47493.png)